### PR TITLE
MAINT: integrate: miscellaneous cleanup

### PR DESCRIPTION
**Pull request by perimosocordiae**
Sorry for the big diff. Here are the highlights:
- removed the `from numpy import ...` namespace pollution
- replaced `np.add.reduce` with `np.sum` and `np.add.accumulate` with `np.cumsum`
- replaced `range` with `xrange` where appropriate

### CODE REVIEW

**Comment by jakevdp** [comment]
Since it's already such a huge diff, I might suggest to also do some PEP8 cleanup as well. A lot of the older scipy library code is not as clean and uniform as the preferred style used today.

**Comment by perimosocordiae** [comment]
@jakevdp For the whole file, or just where I'm making changes?

**Comment by WarrenWeckesser** [comment]
`trapz` has been a public member of the `scipy.integrate` namespace for a long time, and the use of `scipy.integrate.trapz` is pretty common, e.g.
* http://stackoverflow.com/questions/17252024/integrating-using-scipys-trapz-function
* http://stackoverflow.com/questions/30530125/how-to-illustrate-the-simpson-trapezoidal-rules-in-python
* http://stackoverflow.com/questions/20668689/integrating-2d-samples-on-a-rectangular-grid-using-scipy
* http://stackoverflow.com/questions/31964727/integrating-discrete-point-in-python

Removing it will require a deprecation cycle.  I'm not sure it is worth the potential disruption of user's code.  (Heck, it would also be nice to remove the inclusion of `numpy`'s namespace in `scipy`--e.g. `scipy.array`, etc--but that's not going to happen any time soon.)

**Comment by perimosocordiae** [comment]
Ah, that explains the `trapz` situation. I suppose one can't blame users for accessing it through scipy when it's in the scipy.integrate docs. I'll revert the changes that removed it.

**Comment by jakevdp** [comment]
> For the whole file, or just where I'm making changes?

The whole file. Conceptually, I think PEP8 here is in line with the changes you're making (stylistic cleanup based on current community standards). It strikes me as a good time to fix styles across the whole file.

**Comment by perimosocordiae** [comment]
The `trapz` changes have been undone, so the public API hasn't changed at all.

The second commit contains primarily simple PEP8 fixes, though I did make some extra adjustments for readability where I thought it was needed.

**Comment by jakevdp** [comment]
I went through it line-by-line. Much cleaner and easier to read. Thanks for the work on this!

Tests seem to have passed; I'm :+1: for merge. @WarrenWeckesser – are you happy with this?

**Comment by WarrenWeckesser** [comment]
> replaced np.add.reduce with np.sum and np.add.accumulate with np.cumsum

Could you explain why you made this change?  There's nothing wrong with using the `add` methods, and for small arrays it can be significantly faster (since `sum` and `cumsum` ultimately call them anyway).

**Comment by perimosocordiae** [comment]
@WarrenWeckesser I made that change for readability's sake. I didn't do any benchmarking, but I expect that the speed difference will be negligible in this case.

**Comment by jakevdp** [comment]
From my checks, it looks like ``np.sum``/``np.cumsum`` have about a 2 microsecond overhead, FWIW.

**Comment by WarrenWeckesser** [comment]
Well, it probably isn't worth quibbling over.  Overall, this PR is fine.  Thanks for the clean up!

File: scipy/integrate/quadrature.py [hunk 1]
```diff
@@ -379,33 +378,34 @@ def simps(y, x=None, dx=1, axis=-1, even='avg'):
...
     N = y.shape[axis]
     last_dx = dx
     first_dx = dx
     returnshape = 0
     if x is not None:
-        x = asarray(x)
+        x = np.asarray(x)
         if len(x.shape) == 1:
-            shapex = ones(nd)
+            shapex = [1] * nd
```
Comment (larsmans) [inline scipy/integrate/quadrature.py]: If you make this `(1,) * nd`, then you don't need the tuple cast below.

Comment (jakevdp) [inline scipy/integrate/quadrature.py]: But then the next line would fail because tuples are immutable

Comment (larsmans) [inline scipy/integrate/quadrature.py]: Ah, right, hadn't noticed that.

**Comment by larsmans** [comment]
Merged since there there are two +1s here. Thanks @perimosocordiae!

**Pull request merged**